src/baincomplexinstallerdialog.cpp

```diff
@@ -96,9 +96,6 @@ void BainComplexInstallerDialog::updateTree(std::shared_ptr<IFileTree> &tree)
         newTree->merge(entry->astree());
       }
     }
-    else {
-      newTree->insert(entry);
-    }

```

Comment:
The issue was that `entry` was from the `tree` we're iterating on, so it get detached during `insert`, and thus we are modifying the tree while iterating it (same as modifying a vector).

There are other way to fix this issue (`erase` can be used to detach while iterating), but I noticed that in the previous installer version, the files at the root of the archive were not added, so I simply removed this. There are no issue with the `merge` above because `merge(entry->astree())` does not detach `entry` (it simply moves some or all of its children).